src/Repository/CountryRepository.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License (AFL 3.0)
+ * It is available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/afl-3.0.php
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade this file
+ *
+ *  @author    Buckaroo.nl <plugins@buckaroo.nl>
+ *  @copyright Copyright (c) Buckaroo B.V.
+ *  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+namespace Buckaroo\PrestaShop\Src\Repository;
+
+use Buckaroo\PrestaShop\Src\Entity\BkCountries;
+use Doctrine\ORM\EntityRepository;
+
+class CountryRepository extends EntityRepository implements BkCountriesRepositoryInterface
+{
+    private function processCountries($countries)
+    {
+        $result = [];
+
+        foreach ($countries as $country) {
+            $result[] = [
+                'id' => $country['id_country'],
+                'name' => strtolower($country['name']),
+                'iso_code_2' => $country['iso_code'],
+                'iso_code_3' => \Country::getIsoById($country['id_country']),
+                'call_prefix' => $country['call_prefix'],
+                'icon' => \Tools::strtolower($country['iso_code']) . '.jpg',
+            ];
+        }
+
+        return $result;
+    }
+
+    public function getCountriesFromDB(): array
+    {
+        $countries = $this->findAll();
+        $result = [];
+
+        foreach ($countries as $country) {
+            $result[] = [
+                'id' => $country->getCountryId(),
+                'name' => strtolower($country->getName()),
+                'iso_code_2' => $country->getIsoCode2(),
+                'iso_code_3' => $country->getIsoCode3(),
+                'call_prefix' => $country->getCallPrefix(),
+                'icon' => \Tools::strtolower($country->getIsoCode2()) . '.jpg',
+            ];
+        }
+
+        return $result;
+    }
+
+    public function checkAndInsertNewCountries()
+    {
+        $dbCountries = $this->getCountriesFromDB();
+        $dbCountryIds = array_column($dbCountries, 'id');
+
+        $langId = \Context::getContext()->language->id;
+        $rawCountries = \Country::getCountries($langId, true);
```

Comment:
if the countries are in prestashop why do you need to create a new table for countries??

Reply:
I use id`s of saved countries in ordering